lib/Sema/SemaCast.cpp

```diff
@@ -2612,6 +2613,24 @@ void CastOperation::CheckCStyleCast() {
   }
 }
 
+void CastOperation::CheckBoundsCast(BoundsExpr *&Bounds,
```

Comment:
- Bounds does not need to be by-ref parameter.  I think you can just return it.
- The source type can be an integral type or pointer type if this is an assume_bounds_cast.